decentraland/unity-explorer
# fix: SDK input modifier effect persistence after teleport

### WHY

Today if a scene uses the `InputModifier` SDK component to block any of the main player movement interactions, after teleporting somewhere far away for example using the `/goto` chat command, those movement interactions blocking are persisting.

### WHAT

Implemented the missing `IFinalizeWorldSystem` logic in the pertinent system.

-----------

### QA INSTRUCTIONS

1. Download the build from this PR
2. Open it and go to `65,-125` where I've deployed the [input modifier test scene](https://github.com/decentraland/sdk7-test-scenes/tree/main/scenes/0%2C1-input-modifier)
3. In the scene UI click on "disable all" to block all player movement interactions, and confirm that you have those blocked (cannot move, cannot jump)
4. run the following chat command `/goto 100,100`
5. After the teleportation finishes, confirm that you can now move, jump, etc.


https://github.com/user-attachments/assets/b9416c93-5f39-4682-8dfb-22ea2b62a510
